### ISSUE TITLE
build: netbsd-386 builds hang during make.bash

### ISSUE THREAD

**Issue by bradfitz** [golang/go#19339]
The netbsd-386 builds hang during make.bash somewhere.

I'm disabling them from the builders for now, since the build coordinator doesn't yet deal with that well.

/cc @mdempsky @ianlancetaylor @bsiegert

**Comment by bradfitz** [comment]
(These are gomote accessible if anybody wants to reproduce. I'm not disabling that part. The actual host is a NetBSD 7.0 amd64 image, using GOARCH=386 / GOHOSTARCH=386, which gomote will set for you if you create a netbsd-386-70 image)

**Comment by bradfitz** [comment]
I updated the NetBSD builders to NetBSD 7.1.

Maybe something was fixed. We'll see.

/cc @bsiegert

**Comment by bradfitz** [comment]
I guess we should document in the Release Notes' Known Issues section (https://tip.golang.org/doc/go1.9#known_issues) about NetBSD not being stable and kick this to Go 1.10.

Sound okay @mdempsky, @bsiegert, @ianlancetaylor?

**Comment by bradfitz** [comment]
I just attached to the serial console of a netbsd-386-71 builder while running make.bash hoping to catch a NetBSD kernel panic like in #20836 but no luck.

I see nothing to the serial port, but the build hangs at:

```
##### Building packages and commands for netbsd/386.
...
cmd/trace
cmd/vendor/github.com/google/pprof/internal/proftest
cmd/vet/internal/cfg
cmd/vet/internal/whitelist
cmd/vet
                                   
^C
bradfitz@gdev:~/go$ 
bradfitz@gdev:~/go$ gomote run $MOTE go/bin/go version
go version devel gomote.XXXXX netbsd/386
```

Note that the machine is still usable after killing the make.bash run.

Trying it again, it hangs in a different place:

```
##### Building packages and commands for netbsd/386.
...
runtime/race
vendor/golang_org/x/text/secure
vendor/golang_org/x/text/unicode
cmd/compile/internal/test
cmd/vet/internal/whitelist
```

Looking at the GCE console, I see it's using 99% CPU.

![screen shot 2017-06-28 at 12 48 30 pm](https://user-images.githubusercontent.com/2621/27657074-3e7f3b26-5c00-11e7-9c05-934243af0408.png)

**Comment by zoulasc** [comment]
What are the steps to reproduce this?

**Comment by bradfitz** [comment]
@zoulasc, set GOARCH=386, cd into your Go checkout's `src/` directory, and then run `./make.bash` and watch it spin.

We run NetBSD 7.1 VMs on GCE (KVM, virtio, etc), if that matters.

Our image is built using https://github.com/golang/build/tree/master/env/netbsd-amd64

**Comment by bradfitz** [comment]
Benny, can you finish https://go-review.googlesource.com/c/build/+/78619 ?

Let's get netbsd-386 updated to 8.0+

**Comment by bradfitz** [comment]
Actually I think I figured it out.

**Comment by bradfitz** [comment]
The NetBSD i386 kernel never gets networking setup on GCE. See https://github.com/golang/go/issues/20852#issuecomment-347698956

But trying the amd64 kernel with the 32-bit NetBSD binaries with NetBSD 8-BETA, I see the same issue that we saw earlier with NetBSD 7.1: make.bash hangs and just spins CPU.

make.bash normally takes under a minute. After over 15 minutes, I saw:

```
$ gomote run user-bradfitz-netbsd-amd64-8branch-0 /bin/ps -a
 PID TTY   STAT     TIME COMMAND
   2 tty00 Is+   0:00.01 sh /etc/rc autoboot
   8 tty00 I+    0:00.00 sh /etc/rc autoboot
   9 tty00 I+    0:00.00 sh /etc/rc autoboot
  10 tty00 I+    0:00.03 sh /etc/rc autoboot
  11 tty00 S+    0:00.14 sh /etc/rc autoboot
  16 tty00 I+    0:00.00 sh /etc/rc autoboot
  18 tty00 S+    0:00.11 sh /etc/rc autoboot
 371 tty00 I+    0:00.00 sh /etc/rc autoboot
 503 tty00 I+    0:00.00 sh /etc/rc autoboot
 505 tty00 Il+   0:06.14 /buildlet
 590 tty00 Il+   0:01.57 ./cmd/dist/dist bootstrap -a
 931 tty00 Ol+  25:54.12 /tmp/workdir/go/pkg/tool/netbsd_386/go_bootstrap insta
2494 tty00 S+    0:00.00 sleep 3
2495 tty00 O+    0:00.00 /bin/ps -a
```

The go_bootstrap process was spinning or something. I failed to figure out how to ktrace it. But it was consuming CPU.

I saw no disk activity. Or at least no new files being created/appended.

Related: #22914

**Comment by bradfitz** [comment]
I'm now getting successful all.bash runs on netbsd/386 running on a 64-bit host.

I think this was fixed by 70ee9b4a07d356e9187dd23683ed2254c91984cd

Closing.

I'll reopen new bugs if netbsd/386 binaries don't run on 32-bit NetBSD kernels, once tomorrow's NetBSD 32-bit builds are completed with the virtioscsi config fix.
